sdk/resourcemanager/azure-resourcemanager-containerservice/src/test/java/com/azure/resourcemanager/containerservice/KubernetesClustersTests.java

```diff
@@ -530,4 +531,343 @@ public void testFipsEnabled() {
         kubernetesCluster.refresh();
         Assertions.assertTrue(kubernetesCluster.agentPools().get(agentPoolName2).isFipsEnabled());
     }
+
+    @Test
+    public void testUpdateVersion() {
+        KubernetesCluster kubernetesCluster = containerServiceManager.kubernetesClusters()
+            .define(generateRandomResourceName("aks", 15))
+            .withRegion(Region.US_EAST)
+            .withNewResourceGroup(rgName)
+            .withDefaultVersion()
+            .withRootUsername("testaks")
+            .withSshKey(SSH_KEY)
+            .withSystemAssignedManagedServiceIdentity()
+            .defineAgentPool(generateRandomResourceName("ap", 15))
+            .withVirtualMachineSize(ContainerServiceVMSizeTypes.STANDARD_D2_V2)
+            .withAgentPoolVirtualMachineCount(1)
+            .withAgentPoolType(AgentPoolType.VIRTUAL_MACHINE_SCALE_SETS)
+            .withAgentPoolMode(AgentPoolMode.SYSTEM)
+            .attach()
+            .withDnsPrefix(generateRandomResourceName("dns", 15))
+            .create();
+
+        kubernetesCluster.refresh();
+        Assertions.assertEquals("1.27", kubernetesCluster.version());
+
+        kubernetesCluster.update().withVersion("1.28.5").apply();
+        kubernetesCluster.refresh();
+        Assertions.assertEquals("1.28.5", kubernetesCluster.version());
+    }
+
+    @Test
+    public void testCreateKubernetesClusterWithFreePricingTier() {
```

Comment:
Do we need the "create only" tests? Aren't they covered in the following "from to" tests?